script.js

```diff
@@ -1,4 +1,320 @@
+        rank: rankCounter,
+      };
+
+      // Add the new card to the deck
+      cardDeck.push(card);
+
+      // Increment rankCounter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+
+  // Return the completed card deck
+  return cardDeck;
+};
+var deck = makeDeck();
+
+var playerDeck = [];
+var dealerDeck = [];
+var shuffledDeck = [];
+var playervalue = 0;
+var dealervalue = 0;
+var playerDeckforMessage = [];
+var dealerDeckforMessage = [];
+var UserName = "";
+var win = 0;
+var animation = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  console.log(mode);
+
+  if (mode == "name") {
+    mode = "deal";
+    myOutputValue = Name(input);
+  } else if (mode == "deal") {
+    mode = "actualgame";
+    return DealtCards();
+  } else if (mode == "actualgame") {
+    return actualgame(input);
+  } else if (mode == "reward") {
+    return Graphics();
+  }
   return myOutputValue;
 };
+
+var Name = function (input) {
+  if (!(typeof input == "string") || input == 0) {
+    mode = "name";
+    return `Please type in your name.`;
+  } else {
+    shuffledDeck = shuffleCards(deck);
+    UserName = input;
+    return `Hello, ${UserName}. <br><br>The cards have been shuffled. <br><br>Press Submit to deal cards.`;
+  }
+};
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cardDeck array
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+
+var DealtCards = function () {
+  var playerCard1 = shuffledDeck.pop();
+  var playerCard2 = shuffledDeck.pop();
+
+  playerDeck.push(playerCard1);
+  playerDeck.push(playerCard2);
+
+  var dealerCard1 = shuffledDeck.pop();
+  var dealerCard2 = shuffledDeck.pop();
+
+  dealerDeck.push(dealerCard1);
+  dealerDeck.push(dealerCard2);
+
+  playervalue = calculations(playerDeck);
+  dealervalue = calculations(dealerDeck);
+
+  while (dealervalue > 21) {
+    dealerCard2 = shuffledDeck.pop();
+    dealerDeck[1] = dealerCard2;
+    dealerDeckforMessage[1] = ` ${dealerCard2.name} of ${dealerCard2.suit}`;
+
+    dealervalue = calculations(dealerDeck);
+  }
```

Comment:
Kings, Queens and Jacks value suppose to be 10 therefore this code is not necessary once the value have been change in the makeDeck function